velox/core/PlanFragment.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/core/PlanFragment.h"
+#include "velox/core/QueryConfig.h"
+
+namespace facebook::velox::core {
+
+bool PlanFragment::canSpill(const QueryConfig& queryConfig) const {
```

Comment:
This implementation doesn't seem to allow for plan node / operator extensibility, i.e. custom plan nodes cannot specify whether they support spilling or not. I think Meng's suggestion for adding PlanNode::canSpill(config) method will work better.

CC: @spershin @xiaoxmeng